### PR TITLE
Developed Bubble sort.

### PR DESCRIPTION
**Pull request by rootDEV2990**
Please help review my bubble sort in ruby. =)

T.Y.
-Miguel

### CODE REVIEW

File: sortby.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+def sorting(array)
+  sorted = true
+  (0...(array.count - 1)).each do |index|
+    if array[index] > array[index + 1]
+      sorted = false
+      array[index], array[index + 1] = [array[index + 1], array[index]]
+    end
+  end
+  [!sorted, array]
+end
```
Comment (majovanilla) [inline sortby.rb]: In this line, you are returning true or false beside the sorted array. That shouldn't happen if you integrate both methods.

File: sortby.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+def bubble_sort_items(array)
+  items_unsorted = true
+  items_unsorted, array = sorting(array) while items_unsorted
+  array
+end
```
Comment (majovanilla) [inline sortby.rb]: What's this method for? You should only have two methods on this project: 
bubble_sort
bubble_sort_by
I recommend you to change the names to match those.
Also, delete the extra method and make sure you have only two that return the sorted array.

File: sortby.rb (hunk 1 above)
Comment (majovanilla) [inline sortby.rb]: This should be your main method and it should integrate the one on the top. Please make the necessary adjustments.

File: sortby.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    if array[index] > array[index + 1]
+      sorted = false
+      array[index], array[index + 1] = [array[index + 1], array[index]]
+    end
+  end
+  [!sorted, array]
+end
+
+def bubble_sort_by(array)
+  items_unsorted = true
```
Comment (majovanilla) [inline sortby.rb]: This method doesn't take a block as an argument and order the elements using the block. Please make the necessary adjustments to fix it. Right now, when you pass a block, it's returning the array sorted but not with the order of the block passed.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+### Setup
+
+If your using Arch you can type in terminal.. 
+
+sudo pacman -S ruby --noconfirm 
+
+to install. if your OS is diffrent check out the link bellow.
+
+https://www.ruby-lang.org/en/downloads/
```
Comment (majovanilla) [inline README.md:31]: This part is not descriptive of your project. You don't have to use the prerequisites but you should give the user some guidance to run your methods.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hi @rootDEV2990 

## Good job on this project! 👍


There are still some changes 👇 you need to implement to approve this request 🔍

### Required Changes: ✍️

- Please write meaningful pull request messages. Check this [example](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/)

- Check the commented code below.

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Comment by rootDEV2990** [comment]
Rewrote the bubble sort please help review i am still working on fixing rubocop errors

File: sortby.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+def bubble_sort_by(array)
+  loop_size = array.length
+  loop do
+    swap = false
+    (loop_size - 1).times do |i|
+      if yield(array[i], array[i + 1]) > 0
+
+      array[i], array[i + 1] = array[i + 1], array[i]
+      swap = true
```
Comment (suretrust) [inline sortby.rb]: You are missing an `end` here.
```ruby
if (condition)
end
```

File: sortby.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    end
+    break unless swap
+  end
+  array
+end
+
+input = [5, 3, 4, 9, 1, 8]
+p bubble_sort(input)
```
Comment (suretrust) [inline sortby.rb]: Let's remove this once you are done.

File: sortby.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      array[i], array[i + 1] = array[i + 1], array[i]
+      swap = true
+    end
+    break unless swap
+  end
+  array
+end
+
+inp = [1, 3, 4, 5, 8, 9]
+p bubble_sort_by(inp) { |first, second| second - first }
```
Comment (suretrust) [inline sortby.rb]: Let's remove this once you are done.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi, @rootDEV2990, 👋 

### Great job implementing some of the suggested changes. 🙌 Please see my comments on your code. 👇 

- [x] Let's fix all linter errors.

**Kindly submit for another code review request once these issues have been fixed.** _Happy coding!_ 🎉

**Comment by rootDEV2990** [comment]
Hi i added ends to my ifs that reviewer mentioned before but Rubocop "autofixed" it and changed my if to an unless and when i add the end it brakes my code. Do i still need the end? 

Also, I dont know what this means....

Screen Shot 2020-04-29 at 2 55 40 PM

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
The linter file requires changes.

**Comment by pelzolga123** [comment]
## Hi Miguel!  
### Your project looks :+1: amazing!
Just one thing left:
* [x] Make sure your `.rubocop.yml` file has the correct content. You can find original one [here](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml).

### Please kindly fix the above issue and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

**Comment by rootDEV2990** [comment]
Updated rubocop and fixed errors to request approval to merge to master.

**Review by Forison — APPROVED** [review]
## Status:: Approved✔️
## 👋 Miguel 
Good job on addressing the issues raised earlier. Go ahead and merge branch and proceed to the next task
#### Happy coding
Addo

**Pull request merged**